### ISSUE TITLE
Not the correct url value

### ISSUE THREAD

**Issue by long-woo** [axios/axios#2365]
**Describe the bug**

After upgrading to 0.19.0, the `url` obtained in the request interceptor is missing `baseURL`. When I dropped to `0.16.1`, the result was correct.

**To Reproduce**

```js
Axios.defaults.baseURL = '/api'
Axios.interceptors.request.use(
  config => {
    console.log('config', config)
    console.log('config.url', config.url)

    return config
  },
  error => {
    Vue.nprogress.done()

    return error
  }
)
```

**Expected behavior**

Return correctly, bring `baseURL`.

**Environment:**
 - Axios Version [0.19.0]
 - OS: [OSX 10.13.4]
 - Browser [Chrome]
 - Browser Version [76]

**Additional context/Screenshots**

![image](https://user-images.githubusercontent.com/3394035/63321719-1fd17280-c354-11e9-8201-a76576a0123c.png)

**Comment by Lenny-Hu** [comment]
我使用的v1.9也是这种情况，我在拦截器内基于`url`做了一些操作，发现`config.url`并不是以`axios.defaults.baseURL`开头的

**Comment by Alanscut** [comment]
This [breaking changes](https://github.com/axios/axios/blob/master/CHANGELOG.md#0170-oct-21-2017) due to PR #950 which fixed issue #949.  
This commit is used to make the baseURL that rewriting in the interceptor take effect, and the full url should be merged during the request. So `config.url` must be a relative path in interceptor, otherwise we can‘t build the correct full url.

But I don't think this change will affect the operation you want to do, in interceptor, the full url you wanted is `config.baseURL + config.url`, so you can do whatever you want on the `full url` by manipulating the `config.url` and `config.baseURL`.

**Comment by Alanscut** [comment]
`config.url` without `baseURL` could avoid repeatedly adding `baseURL` to the front of requestURL when retry the request. Detailed description: #1628, #2391
